javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserClassDeclaration.java

```diff
@@ -148,9 +147,15 @@ public boolean isStatic() {
                                 public ResolvedTypeDeclaration declaringType() {
                                     return f.declaringType();
                                 }
+                                
+                                @Override
+                                public Optional<FieldDeclaration> toAst() {
+                                    FieldDeclaration fd = JavaParserFieldDeclaration.class.isAssignableFrom(f.getClass()) ?
+                                            ((JavaParserFieldDeclaration)f).getWrappedNode() : null;
+                                    return Optional.ofNullable(fd);
```

Comment:
I think this should be `return f.toAst();` instead of the code here.
In addition, you should implement the `toAst()` method in the class `JavaParserFieldDeclaration` and return the wrappedNode there.

Reply:
Yes it's really better. Thanks for the review.